platform-sdk/swirlds-common/src/main/java/com/swirlds/common/metrics/platform/prometheus/AbstractMetricAdapter.java

```diff
@@ -40,4 +73,83 @@ public int incAndGetReferenceCount() {
     public int decAndGetReferenceCount() {
         return referenceCount.decrementAndGet();
     }
+
+    /**
+     * A holder for common metric values supporting automatic naming rules conversions. Instances of this class are
+     * created based on a {@code Metric}, and naming rules conversions will be applied to the category, name, and unit
+     * fields according to {@link NameConverter}.
+     *
+     * <p>
+     * The class facilitates the creation of Prometheus-compatible metric values by storing the fixed and converted
+     * category, name, unit, and help fields.
+     * </p>
+     */
+    private static class ConvertedMetricValues {
+        private final boolean unitless;
+
+        @NonNull
+        private final String subSystem;
+
+        @NonNull
+        private final String name;
+
+        @NonNull
+        private final String unit;
+
+        @NonNull
+        private final String help;
+
+        private ConvertedMetricValues(boolean unitless, final @NonNull Metric metric) {
+            Objects.requireNonNull(metric, "metric must not be null");
+            this.unitless = unitless;
```

Comment:
I do not get that flag. Metric defines the type field. Why not just check if that field is null?

Reply:
The problem is that prometheus client will fail if we set a unit to a metric that should not have a unit set.
But our code doesn't prevent this from happening; we allow users to set units in any metric type even when Prometheus fails (we have one occurrence that sets "boolean" and other values in tests), which makes it impossible to rely only on a check against null. What was happening before is that the adapters knew not to set the unit field when necessary. I needed that information in the superclass instead of each adapter.
I guess that the final fix would be not to allow to set units on invalid metrics, and we could remove the flag, just that the change was way bigger.

Reply:
ok, if I understood it correctly "supportsUnit" would be a better param from my point of view. That one would define if the metric supports a unit. Next to that the Javadoc is missing that would help to understand the param

Reply:
Agreed, `supportsUnit` sounds better. I had the intention that the default false value set the unit instead of not setting it, but `unitless` ended up being a strange name. 
I didn't add Javadoc in this method given that it is a private method on a private inner class, the inner class has a Javadoc, and the containing class also has one explaining the receiving parameters. I'll add one here too, but I'm checking to be on the same page for the future: Should we always add Javadoc or only in public interfaces?